test-support/helpers/ember-tooltips.js

```diff
@@ -33,6 +75,43 @@ const getTooltipTargetFromBody = function(selector = tooltipOrPopoverTargetSelec
   return $tooltipTarget;
 }
 
+function getOppositeSide(side) {
+  switch (side) {
+    case 'top': return 'bottom'; break;
+    case 'right': return 'left'; break;
+    case 'bottom': return 'top'; break;
+    case 'left': return 'right'; break;
+  }
+}
+
+function validateSide(side, testHelper = 'assertTooltipSide') {
+  const sideIsValid = (
+    side === 'top' ||
+    side === 'right' ||
+    side === 'bottom' ||
+    side === 'left'
+  );
+
+  /* We make sure the side being tested is valid. We
+  use Ember.assert because assert is passed in from QUnit */
+
+  if (!sideIsValid) {
+    Ember.assert(`You must pass side like ${testHelper}(assert, { side: 'top' }); Valid options for side are top, right, bottom, and left.`);
+  }
+}
+
+function getTooltipPosition(options = {}) {
+  const $target = getTooltipTargetFromBody(options.targetSelector);
+  const targetPosition = $target[0].getBoundingClientRect();
```

Comment:
@a15n With reference to your question about why `getBoundingClientRect()` is better than jQuery's `position()` method, using `getBoundingClientRect()` avoids [this mess](https://github.com/sir-dunxalot/ember-tooltips/blob/01991d37bd1ad0c31da89606432e4c8d57a69321/tests/integration/components/spacing-test.js#L11).